feat: add dashboard modal and map filters dropdown (UI only, no integintegration)

TiagoBernardoSantos opened the pull request:
Descrição do PR

Este PR realiza melhorias na Home da aplicação:

Move o dashboard de uma tela a parte para a home, acessível via modal.

Ajusta os filtros do mapa para um botão com dropdown, facilitando a interação do usuário.

💡 Observação: Essa alteração é apenas visual; ainda não há integração com o backend.

[Card](https://denariusdata2024.atlassian.net/browse/RAD-81?atlOrigin=eyJpIjoiZTEyMzczMzA0NjliNDhmYzk0ZmZhY2U2MzRmNTg4NzciLCJwIjoiaiJ9) 

### Imagens: 

### Home
<img width="1366" height="641" alt="image" src="https://github.com/user-attachments/assets/e5406ba0-aeba-4bcf-a998-dfc011e011ba" />


### Filtro selecionado
<img width="1365" height="641" alt="image" src="https://github.com/user-attachments/assets/aff05bb1-6dcf-49da-bf52-35f5131872f6" />

### Dashboard
<img width="1366" height="640" alt="image" src="https://github.com/user-attachments/assets/584daf25-bf16-4762-8aa6-0cbc95ce3ffc" />

TiagoBernardoSantos committed:
feat: add dashboard modal and map filters dropdown (UI only, no integration)

augustopiatto commented on src/modules/protocols/ProtocolsView.vue:
```diff
@@ -9,11 +9,11 @@ import type {
   ModalType,
   TabType
-} from '@/modules/protocols/types/ProtocolsTypes';
+} from '@/modules/protocols/types/protocolsTypes';
```
toma cuidado que aqui eu corrigi, parece q vc ta tirando as mudanças.
Lembra de atualizar sua branch com a sprint-3:
git checkout sprint-3
git pull
git checkout RAD-81/DashBoards-main-road-data
git pull
git merge sprint-3 -m "chore: update with sprint-3"

E ai resolve os conflitos

augustopiatto commented on src/modules/home/HomeView.vue:
```diff
@@ -278,24 +583,38 @@ function clearSelection() {
+  }
+}
 </style>
+add dashboard modal and map filters dropdown (UI only, no integration)
```
esse texto ficou solto, era alguma mensagem de commit ou algo assim? Acho q digitou no lugar errado

augustopiatto commented on src/modules/dashboards/DashboardsView.vue:
```diff
@@ -1,86 +1,58 @@
-
 <script lang="ts" setup>
-import { ref, onMounted, watch } from 'vue'
+import { ref, onMounted } from 'vue'
```
recomendo vc mudar o nome do arquivo, ele não é mais uma "view". Quando um arquivo tem nome "View" é pq ele tem uma url pra chegar nele. Nesse caso vc removeu o "/dashboard", e isso aqui virou um popup. Eu mudaria pra "DashboardsPopup"

TiagoBernardoSantos committed:
fix: resolve merge conflicts

TiagoBernardoSantos committed:
feat: add comparison button, zones carousel and improve line chart UI

- Implemented new comparison functionality - Added zones carousel to enhance navigation - Improved layout and usability of the line chart

TiagoBernardoSantos committed:
chore: update with sprint-3

TiagoBernardoSantos committed:
chore: update with sprint-3

TiagoBernardoSantos committed:
chore: update with sprint-3

TiagoBernardoSantos committed:
feat: Implementation of the front-end and back-end integration for filters and dashboards, and implementation of the DashboardPopups button.

augustopiatto committed:
chore: update with sprint-3

augustopiatto committed:
refactor: home filter component with sprint-3 UI

augustopiatto committed:
refactor: dashboard alert position

augustopiatto committed:
fix: canvas rendering

augustopiatto committed:
refactor: button position
